tests/test_cmdLineFwk.py

```diff
@@ -0,0 +1,216 @@
+    def loadTaskClass(self, taskName):
+        if taskName == "TaskOne":
+            return TaskOne, "TaskOne"
+        elif taskName == "TaskTwo":
+            return TaskTwo, "TaskTwo"
+
+    def makeTask(self, taskClass, config, overrides, butler):
+        if config is None:
+            config = taskClass.ConfigClass()
+            if overrides:
+                overrides.applyTo(config)
+        return taskClass(config=config, butler=butler)
+
+
+def _makeArgs(pipeline=None, qgraph=None, pipeline_actions=(), order_pipeline=False,
+              save_pipeline="", save_qgraph="", pipeline_dot="", qgraph_dot=""):
+    """Return parsed command line arguments.
+
+    Parameters
+    ----------
+    pipeline : `str`, optional
+        Name of the pickle file with pipeline.
+    qgraph : `str`, optional
+        Name of the pickle file with QGraph.
+    pipeline_actions : itrable of `cmdLinePArser._PipelineAction`, optional
+    order_pipeline : `bool`
+    save_pipeline : `str`
+        Name of the pickle file to store pipeline.
+    save_qgraph : `str`
+        Name of the pickle file to store QGraph.
+    pipeline_dot : `str`
+        Name of the DOT file to write pipeline graph.
+    qgraph_dot : `str`
+        Name of the DOT file to write QGrpah representation.
+    """
+    args = argparse.Namespace()
+    args.pipeline = pipeline
+    args.qgraph = qgraph
+    args.pipeline_actions = pipeline_actions
+    args.order_pipeline = order_pipeline
+    args.save_pipeline = save_pipeline
+    args.save_qgraph = save_qgraph
+    args.pipeline_dot = pipeline_dot
+    args.qgraph_dot = qgraph_dot
+    return args
+
+
+class CmdLineFwkTestCase(unittest.TestCase):
+    """A test case for CmdLineFwk
+    """
+
+    def testMakePipeline(self):
+        """Tests for CmdLineFwk.makePipeline method
+        """
+        fwk = CmdLineFwk()
+        taskFactory = TaskFactoryMock()
+
+        # make empty pipeline
+        args = _makeArgs()
+        pipeline = fwk.makePipeline(taskFactory, args)
+        self.assertIsInstance(pipeline, Pipeline)
+        self.assertEqual(len(pipeline), 0)
+
+        # few tests with pickle
+        with makeTmpFile() as tmpname:
+            # make empty pipeline and store it in a file
+            args = _makeArgs(save_pipeline=tmpname)
+            pipeline = fwk.makePipeline(taskFactory, args)
+            self.assertIsInstance(pipeline, Pipeline)
+
+            # read pipeline from a file
+            args = _makeArgs(pipeline=tmpname)
+            pipeline = fwk.makePipeline(taskFactory, args)
+            self.assertIsInstance(pipeline, Pipeline)
+            self.assertEqual(len(pipeline), 0)
+
+            # pickle with wrong object type
+            with open(tmpname, "wb") as pickleFile:
+                pickle.dump({}, pickleFile)
+            args = _makeArgs(pipeline=tmpname)
+            with self.assertRaises(TypeError):
+                pipeline = fwk.makePipeline(taskFactory, args)
```

Comment:
Small nitpick, but a statement assigning this to a variable seems needless for this and the test below.